Samples/Starbucks-sample/Starbucks-sample/Overlay/Results/ResultsViewController.h

```diff
@@ -0,0 +1,27 @@
+//
+//  ResultsViewController.h
+//  Starbucks-sample
+//
+//  Created by Jure Cular on 29/06/2017.
+//  Copyright © 2017 MicroBlink. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@protocol ResultsViewControllerDelegate;
+
+@interface ResultsViewController : UIViewController
+
+- (instancetype)initWithLabelsMap:(NSDictionary<NSString *, NSString *> *)labelsMap;
```

Comment:
If the ViewController has only one initializer which should be used (using others would be an error), then these initializer should be marked with NS_DESIGNATED_INITIALIZER. All other intnializers (such as initWithNibNamed: ) should be marked with NS_UNAVAILABLE.

https://developer.apple.com/library/content/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html

https://www.codementor.io/tips/7142988439/disallow-objective-c-method-calls-with-compile-time-errors-via-ns_unavailable